proposals/p5270.md

```diff
@@ -0,0 +1,152 @@
+issues[^1], as tools like grep or clang-tidy do not differentiate our active
+toolchain codebase from the frozen explorer one.
+
+As we update Bazel, Clang, Clang-tidy, etc, we also have to update the explorer
+codebase to keep it building cleanly.
+
+As we modify the `//testing` harness, we have to accommodate and work around the
+explorer tests[^2][^3][^4].
+
+[^1]: https://github.com/carbon-language/carbon-lang/pull/5224
+
+[^2]: https://github.com/carbon-language/carbon-lang/pull/4979
+
+[^3]: https://github.com/carbon-language/carbon-lang/pull/5025
+
+[^4]: https://github.com/carbon-language/carbon-lang/pull/5036
+
+## Background
+
+-   Previous proposal:
+    [#3532: Focus implementation effort on the toolchain](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p3532.md)
+-   Kick-off of this discussion
+    [#5224](https://github.com/carbon-language/carbon-lang/pull/5224#pullrequestreview-2730512195)
+-   Discord discussion:
+    [2025-04-02 #explorer](https://discord.com/channels/655572317891461132/763516049710120960/1356979197070803095)
+
+## Proposal
+
+1. Add a tag `explorer-archived` in the main `carbon-lang` git repository.
+2. Create a new `explorer` repository under the `carbon-language` organization
+   that only contains the `//explorer` and `//installers` directories and their
+   dependencies at head.
+3. Locally ensure the explorer tests build and pass under the `explorer`
+   repository.
+4. Add a `README.md` to the `explorer` repository that explains explorer is
+   archived and not under active development.
+5. Remove the "Explorer (trunk)" compiler option from
+   [carbon.compiler-explorer.com](https://carbon.compiler-explorer.com)
+6. Delete `//explorer` and `//installers` in the main `carbon-lang` repository.
+7. Archive the `explorer` repository in GitHub, making it read-only.
+
+Note that fuzzer test cases from the explorer are already relocated under
+`//toolchain/*/fuzzer_corpus/`.
+
+## Rationale
+
+The primary purpose of the explorer codebase at this time is as a demonstration
+of past work implementing the carbon language design. This can help to inform
+the implementation of the toolchain as it catches up in various areas, as long
+as the design has not deviated from the explorer implementation.
+
+Searching in, and providing links to the explorer codebase comes up in design
+discussions occasionally [^5][^6][^7][^8][^9][^10], and we should maintain the
+ability of toolchain developers to look through the explorer easily. The primary
+place they do so is on GitHub, where the code can be linked to. GitHub search
+only works
+[on the main branch](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#using-qualifiers)
+of a repository, so the `trunk` branch for the `carbon-lang` repository. To
+maintain searchability, the explorer codebase must either remain on `trunk` in
+`carbon-lang` or in a sibling repository.
```

Comment:
Might be worth noting here that github search [does appear to work in archived repositories](https://github.com/search?q=org%3Acarbon-language+InstructionType_conditionalJmpInst&type=code).